Content.Shared/Damage/Systems/StaminaSystem.cs

```diff
@@ -219,17 +222,27 @@ private void OnProjectileHit(EntityUid uid, StaminaDamageOnCollideComponent comp
 
     private void OnThrowHit(EntityUid uid, StaminaDamageOnCollideComponent component, ThrowDoHitEvent args)
     {
-        OnCollide(uid, component, args.Target);
+        if (TryComp<ThrownItemComponent>(uid, out var thrownComp) &&
+            HasComp<StaminaComponent>(args.Target))
+        {
+            if (OnCollide(uid, component, args.Target))
+            {
+                _thrownItemSystem.StopThrow(uid, thrownComp);
+            }
+        }
     }
 
-    private void OnCollide(EntityUid uid, StaminaDamageOnCollideComponent component, EntityUid target)
+    private bool OnCollide(EntityUid uid, StaminaDamageOnCollideComponent component, EntityUid target)
     {
+        if (!TryComp<StaminaComponent>(target, out var staminaComp))
+            return false;
+
         var ev = new StaminaDamageOnHitAttemptEvent();
         RaiseLocalEvent(uid, ref ev);
         if (ev.Cancelled)
-            return;
+            return false;
 
-        TakeStaminaDamage(target, component.Damage, source: uid, sound: component.Sound);
+        return TakeStaminaDamage(target, component.Damage, staminaComp, uid, sound: component.Sound);
```

Comment:
You're resolving staminacomponent twice which is unnecessary, just pass it into the method as it's far cheaper.

Also call it TryCollide instead if you're changing the return type.